### DISCUSSION TITLE
Creating mask for gigapixel images using pyvips

### DISCUSSION THREAD

**Discussion by Rasaa84** [libvips/libvips#3301] [Q&A]
Hi,

I am working on whole-slide images and need to create a mask highlighting some part of the image. It can be done using opencv as follows:

```
mask = cv2.resize(mask, (slide.shape[1], slide.shape[0]))
mask = np.uint8(mask * 255)
mask = cv2.applyColorMap(mask, cv2.COLORMAP_JET)
overlaid_img = mask * 0.3 + slide
```

The issue with this piece of code is that the image needs to be loaded in memory which is problematic working with whole slide images. I am wondering if this can be handle using pyvips without loading the image to memory.

Thanks in advance for your help.

**Comment by jcupitt** [comment]
Hi @Rasaa84,

Sure, it's very like your opencv sample. In python:

```python
#!/usr/bin/python3

import sys
import pyvips

if len(sys.argv) != 4:
    print("usage: mask.py input-slide output-slide mask")
    sys.exit(1)

slide = pyvips.Image.new_from_file(sys.argv[1])
mask = pyvips.Image.new_from_file(sys.argv[3])
mask = mask.resize(slide.width / mask.width,
                   vscale=slide.height / mask.height)
slide = (slide + 0.3 * (mask * 255).falsecolour()).cast("uchar")
slide.write_to_file(sys.argv[2])
```

I can run:

```
$ ./mask.py ~/pics/openslide/CMU-1.svs[rgb] x.tif[tile,compression=jpeg,pyramid] mask.v
```

To read an sys and write a tiled pyramidal jpeg tiff.

**Reply by Rasaa84** [reply]
Thanks @jcupitt  for your response.

How can I check if an images has an alpha channel? and how can I get ride of alpha channel?

**Reply by jcupitt** [reply]
```python
if image.hasalpha():
    # take just RGB
    image = image[0:3]
```
